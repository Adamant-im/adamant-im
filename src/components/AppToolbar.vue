<template>
  <v-toolbar
    :flat="flat"
    :height="height"
    app
  >
    <v-btn v-if="showBack" @click="goBack" icon>
      <v-icon>mdi-arrow-left</v-icon>
    </v-btn>

    <v-toolbar-title v-if="title">
      <div>{{ title }}</div>
      <div v-if="subtitle" class="body-1">{{ subtitle }}</div>
    </v-toolbar-title>
  </v-toolbar>
</template>

<script>
export default {
  methods: {
    goBack () {
      this.$router.back()
    }
  },
  props: {
    title: {
      type: String,
      default: undefined
    },
    subtitle: {
      type: String,
      default: undefined
    },
    flat: {
      type: Boolean,
      default: false
    },
    height: {
      type: Number,
      default: 64
    },
    showBack: {
      type: Boolean,
      default: true
    }
  }
}
</script>
