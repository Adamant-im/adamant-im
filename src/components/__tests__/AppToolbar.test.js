import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'

import AppToolbar from '@/components/AppToolbar'

Vue.use(Vuetify)

describe('AppToolbar.vue', () => {
  it('renders the correct markup', () => {
    const wrapper = shallowMount(AppToolbar, {
      mocks: {
        $router: {
          back: jest.fn()
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should call $router.back() when goBack()', () => {
    const wrapper = shallowMount(AppToolbar, {
      mocks: {
        $router: {
          back: jest.fn()
        }
      }
    })

    wrapper.vm.goBack()
    expect(wrapper.vm.$router.back).toHaveBeenCalled()
  })

  it('should render with props', () => {
    const wrapper = shallowMount(AppToolbar, {
      propsData: {
        title: 'Title',
        subtitle: 'Subtitle'
      },
      mocks: {
        $router: {
          back: jest.fn()
        }
      }
    })

    expect(wrapper.html()).toContain('<div>Title</div>')
    expect(wrapper.html()).toContain('<div class="body-1">Subtitle</div>')
  })
})
