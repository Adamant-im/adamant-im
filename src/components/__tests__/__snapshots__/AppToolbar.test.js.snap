// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolbar.vue renders the correct markup 1`] = `
<v-toolbar-stub
  app=""
  height="64"
  scrollthreshold="300"
>
  <v-btn-stub
    activeclass="v-btn--active"
    icon="true"
    tag="button"
    type="button"
  >
    <v-icon-stub>
      mdi-arrow-left
    </v-icon-stub>
  </v-btn-stub>
   
  <!---->
</v-toolbar-stub>
`;
